File: examples/junit5/junit5-gradle/src/test/java/org/example/ParametrisedTests.java
package org.example;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

public class ParametrisedTests {
    static Stream<Arguments> arguments() {
        return Stream.of(
                Arguments.of("string1"),
                Arguments.of("string2"),
                Arguments.of("string3")
        );
    }

    @ParameterizedTest
    @MethodSource("arguments")
    public void testMethod_success1(String data) {
        System.out.println("Data is: " + data);
    }

    @ParameterizedTest
    @MethodSource("arguments")
    public void testMethod_failed1(String data) {
        System.out.println("Data is: " + data);
        throw new RuntimeException("Test failed");
    }
}

File: examples/kotlin/junit5/src/test/kotlin/ParametrisedTests.kt
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

class ParametrisedTests {

    companion object {
        @JvmStatic
        fun arguments(): Stream<Arguments> = Stream.of(
            Arguments.of("string1"),
            Arguments.of("string2"),
            Arguments.of("string3")
        )
    }

    @ParameterizedTest
    @MethodSource("arguments")
    fun testMethod_success1(data: String) {
        println("Data is: $data")
    }

    @ParameterizedTest
    @MethodSource("arguments")
    fun testMethod_failed1(data: String) {
        println("Data is: $data")
        throw RuntimeException("Test failed")
    }
}
